File: DTMS365-AppSource/DynamicsTMS365/Models/BookNote.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DynamicsTMS365.Models
{
    //Added By LVV on 6/23/20 for v-8.2.1.008 Task #20200609162832 - Create Book Notes page and add Navigation item to the Load Board
    public class BookNote
    {
        public int BookNotesControl { get; set; }
        public int BookNotesBookControl { get; set; }
        public string BookNotesVisable1 { get; set; }
        public string BookNotesVisable2 { get; set; }
        public string BookNotesVisable3 { get; set; }
        public string BookNotesVisable4 { get; set; }
        public string BookNotesVisable5 { get; set; }
        public string BookNotesConfidential1 { get; set; }
        public string BookNotesConfidential2 { get; set; }
        public string BookNotesConfidential3 { get; set; }
        public string BookNotesConfidential4 { get; set; }
        public string BookNotesConfidential5 { get; set; }
        public string BookNotesBookUser1 { get; set; }
        public string BookNotesBookUser2 { get; set; }
        public string BookNotesBookUser3 { get; set; }
        public string BookNotesBookUser4 { get; set; }

        public DateTime? BookNotesModDate { get; set; }

        private string _BookNotesModUser;
        public string BookNotesModUser
        {
            get { return _BookNotesModUser.Left(100); } //uses extension string method Left
            set { _BookNotesModUser = value.Left(100); }
        }

        private byte[] _BookNotesUpdated;

        /// <summary>
        /// BookNotesUpdated should be bound to UI, _BookNotesUpdated is only bound on the controller
        /// </summary>
        public string BookNotesUpdated
        {
            get
            {
                if (this._BookNotesUpdated != null) { return Convert.ToBase64String(this._BookNotesUpdated); }
                return string.Empty;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) { this._BookNotesUpdated = null; } else { this._BookNotesUpdated = Convert.FromBase64String(value); }
            }
        }

        public void setUpdated(byte[] val) { _BookNotesUpdated = val; }
        public byte[] getUpdated() { return _BookNotesUpdated; }

        public string BookSHID { get; set; }
        public string BookConsPrefix { get; set; }
        public string BookProNumber { get; set; }    
        public string BookCarrOrderNumber { get; set; }
        public int BookOrderSequence { get; set; }
    }
}

File: NGL.DataAssemblies/Ngl.FreightMaster.Data/DTO/BookData/BookNote.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Runtime.Serialization
Imports Ngl.Core.ChangeTracker

Namespace DataTransferObjects
    <DataContract()> _
    Public Class BookNote
        Inherits DTOBaseClass


#Region " Data Members"
        Private _BookNotesControl As Integer = 0
        <DataMember()> _
        Public Property BookNotesControl() As Integer
            Get
                Return _BookNotesControl
            End Get
            Set(ByVal value As Integer)
                _BookNotesControl = value
            End Set
        End Property

        Private _BookNotesBookControl As Integer = 0
        <DataMember()> _
        Public Property BookNotesBookControl() As Integer
            Get
                Return _BookNotesBookControl
            End Get
            Set(ByVal value As Integer)
                _BookNotesBookControl = value
            End Set
        End Property

        Private _BookNotesVisable1 As String = ""
        <DataMember()> _
        Public Property BookNotesVisable1() As String
            Get
                Return Left(_BookNotesVisable1, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesVisable1 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesVisable2 As String = ""
        <DataMember()> _
        Public Property BookNotesVisable2() As String
            Get
                Return Left(_BookNotesVisable2, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesVisable2 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesVisable3 As String = ""
        <DataMember()> _
        Public Property BookNotesVisable3() As String
            Get
                Return Left(_BookNotesVisable3, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesVisable3 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesVisable4 As String = ""
        <DataMember()> _
        Public Property BookNotesVisable4() As String
            Get
                Return Left(_BookNotesVisable4, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesVisable4 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesVisable5 As String = ""
        <DataMember()> _
        Public Property BookNotesVisable5() As String
            Get
                Return Left(_BookNotesVisable5, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesVisable5 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesConfidential1 As String = ""
        <DataMember()> _
        Public Property BookNotesConfidential1() As String
            Get
                Return Left(_BookNotesConfidential1, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesConfidential1 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesConfidential2 As String = ""
        <DataMember()> _
        Public Property BookNotesConfidential2() As String
            Get
                Return Left(_BookNotesConfidential2, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesConfidential2 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesConfidential3 As String = ""
        <DataMember()> _
        Public Property BookNotesConfidential3() As String
            Get
                Return Left(_BookNotesConfidential3, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesConfidential3 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesConfidential4 As String = ""
        <DataMember()> _
        Public Property BookNotesConfidential4() As String
            Get
                Return Left(_BookNotesConfidential4, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesConfidential4 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesConfidential5 As String = ""
        <DataMember()> _
        Public Property BookNotesConfidential5() As String
            Get
                Return Left(_BookNotesConfidential5, 250)
            End Get
            Set(ByVal value As String)
                _BookNotesConfidential5 = Left(value, 250)
            End Set
        End Property

        Private _BookNotesBookUser1 As String = ""
        <DataMember()> _
        Public Property BookNotesBookUser1() As String
            Get
                Return Left(_BookNotesBookUser1, 4000)
            End Get
            Set(ByVal value As String)
                _BookNotesBookUser1 = Left(value, 4000)
            End Set
        End Property

        Private _BookNotesBookUser2 As String = ""
        <DataMember()> _
        Public Property BookNotesBookUser2() As String
            Get
                Return Left(_BookNotesBookUser2, 4000)
            End Get
            Set(ByVal value As String)
                _BookNotesBookUser2 = Left(value, 4000)
            End Set
        End Property

        Private _BookNotesBookUser3 As String = ""
        <DataMember()> _
        Public Property BookNotesBookUser3() As String
            Get
                Return Left(_BookNotesBookUser3, 4000)
            End Get
            Set(ByVal value As String)
                _BookNotesBookUser3 = Left(value, 4000)
            End Set
        End Property

        Private _BookNotesBookUser4 As String = ""
        <DataMember()> _
        Public Property BookNotesBookUser4() As String
            Get
                Return Left(_BookNotesBookUser4, 4000)
            End Get
            Set(ByVal value As String)
                _BookNotesBookUser4 = Left(value, 4000)
            End Set
        End Property

        Private _BookNotesModDate As System.Nullable(Of Date)
        <DataMember()> _
        Public Property BookNotesModDate() As System.Nullable(Of Date)
            Get
                Return _BookNotesModDate
            End Get
            Set(ByVal value As System.Nullable(Of Date))
                _BookNotesModDate = value
            End Set
        End Property

        Private _BookNotesModUser As String = ""
        <DataMember()> _
        Public Property BookNotesModUser() As String
            Get
                Return Left(_BookNotesModUser, 100)
            End Get
            Set(ByVal value As String)
                _BookNotesModUser = Left(value, 100)
            End Set
        End Property

        Private _BookNotesUpdated As Byte()
        <DataMember()> _
        Public Property BookNotesUpdated() As Byte()
            Get
                Return _BookNotesUpdated
            End Get
            Set(ByVal value As Byte())
                _BookNotesUpdated = value
            End Set
        End Property
#End Region

#Region " Public Methods"
        Public Overrides Function Clone() As DTOBaseClass
            Dim instance As New BookNote
            instance = DirectCast(MemberwiseClone(), BookNote)
            Return instance
        End Function

#End Region

    End Class
End Namespace